### DISCUSSION TITLE
Lost all beatmaps on Lazer and an old ver. of Lazer after update 327.0

### DISCUSSION THREAD

**Discussion by bearsomehand** [ppy/osu#17504] [Q&A] [answered]
### Type

Other

### Bug description

I have stable, newest lazer and lazer 2021.320.0 on my pc. I DLed the previous ver. 2 days ago I think, or the day before that and yesterday I was playing around with it a bunch. Did some multi in the evening then watched a replay and then shut it down. Opened it up again because I remembered I wanted to check something and then I got an update(I don't remember if I closed/opened it another time after that).

I then went to my maps and it says I have no maps(I imported all my 20ish gb of maps from stable 2 days ago or the day before that) and asks me if I want to import or start fresh again. It still has my skins. Stable still has the maps but my other, older, lazer also lost all its maps(the reason I saved that ver. was because I had some replays there that got erased from stable when I updated a couple maps). I didn't touch that client yesterday(before checking it) but it still had its maps deleted..?

I didn't notice anything at all that would clue me in that so many maps were being deleted and my pc is pretty old and slow.

I can't tell by disc space if the maps were deleted from the drive or not because I don't keep track(I can fit 40gb more rn) but the folders that both lazer ver. shortcuts point to are about 633mb.

I'm about 95% certain I had stable up when lazer was updated.

The old lazer ver. still says it's on 2021.320.0. I tried to update that ver. before I installed the second to newest ver. seperately but nothing happened when I tried to in-game. When I clicked the update file in the folder a window opened for a fraction of a second and the Squirrel setup txt file wrote this(one line per time I tried):

2022-03-26 12:59:56> Program: Starting Squirrel Updater: 
2022-03-26 13:00:01> Program: Starting Squirrel Updater: 
2022-03-26 13:02:17> Program: Starting Squirrel Updater: 
2022-03-26 13:02:18> Program: Starting Squirrel Updater: 
2022-03-26 13:02:19> Program: Starting Squirrel Updater: 

I posted this on the forum help section and someone told me to post it here...
I'm handsomebear on osu!

### Version

osu! version: 2022.327.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/8361700/database.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/8361701/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/8361702/network.log)
[performance.log](https://github.com/ppy/osu/files/8361703/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/8361704/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/8361705/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/8361706/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/8361707/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8361708/updater.log)

**Comment by peppy** [comment, marked as answer]
Please check your osu! folder for a `backups` folder and look for any files in there. osu! will always make a backup. If you attach the latest version you have of `client.db` (may contain a suffix) we can look into the issue further.

**Reply by bearsomehand** [reply]
Backups(not sure how to read it or what to do with it):
![Backups](https://user-images.githubusercontent.com/102517196/160392337-34adad03-ba99-4f6c-8e6e-bb02276ef9ff.png)

Github isn't letting me attach the client.db(from C:\Users\[name]\AppData\Roaming\osu) files raw but one of the suggested formats is ZIP so I'll try that:
[client.zip](https://github.com/ppy/osu/files/8362390/client.zip)

I'm not very tech savvy if that isn't clear ( :

**Reply by bearsomehand** [reply]
To be clear, I'm perfectly happy(and have the time-) to go through with any troubleshooting etc if you think it can illuminate some issue or be helpful in any way. 

But the 2 biggest issues for me is losing the replays in my old lazer version(I can live with that if it's a hassle or impossible to recover) and more importantly not wanting this to happen again as I want to move over to lazer completely. I can live with just uninstalling both my lazers, starting fresh, re-importing my stable maps and going from there if you think this is a waste of time and/or extremely unlikely to happen again.

I installed the old ver. in "C:\Program Files (x86)"(this is where I have stable as well) and can't find any backup folder there. The new ver. backup folder is under roaming while the rest is in local. I don't really have any experience with any game using the kind of folder system lazer does so it's confusing to me.

**Reply by peppy** [reply]
You did good, but could you please also attach the top file in that screenshot (also zipped)? The 6,324kb one. It should definitely be enough to recover all your data.

**Reply by bearsomehand** [reply]
[client.before_final_migration_1648296190.zip](https://github.com/ppy/osu/files/8363246/client.before_final_migration_1648296190.zip)

**Reply by peppy** [reply]
Make sure osu! is closed then unzip and replace the existing `client.realm` in the main directory (not the backup one). You should have all your old beatmaps and scores back!

[client.realm.zip](https://github.com/ppy/osu/files/8363418/client.realm.zip)

**Reply by bearsomehand** [reply]
I did as you said and first it didn't work. I backed up the old client.realm so I tried doing it 2 more times and then all of the sudden it worked. The maps on my old lazer client was in the new one but I'm getting a couple "beatmap failed to load" errors when I go to map select. When I tried to replay one of the really old plays nothing happened, then I tried another and it crashed.

So I went to check the old lazer client and there were no beatmaps there at all still. Went back to the new one and now it also says there are no beatmaps again. So I went back to the old client.realm and extracted the new again and the old maps are back again. Getting "beatmap failed to load" errors. 

I tried clicking on another replay this time and nothing happened the first 2 times then it crashed again. When I re-open the maps are still there, so it seems it bugs out when I open the old version. All map diffs are 0 stars and when I try to play a map live it starts loading but then just stops and goes back to map select. Video showing both: https://www.youtube.com/watch?v=0CNbvIETmVQ

Also I went to check uninstall apps before and there's only stable and one lazer there.

**Reply by peppy** [reply]
Sounds like somehow the underlying files are missing. No idea how this happened, and unfortunately your lots don't help understand why. You can try importing from stable again (while your client is in the "beatmap failed to load" state) from settings (scroll right to the bottom) and hope for the best. At very least you will be able to see your scores, but likely not be able to view replays.

**Reply by bearsomehand** [reply]
Alright. I'm just going to start fresh with Lazer then, scrub -> reinstall and see how it goes, I only wanted the old replays to make fun progression type videos with anyway(it's on GY maps that were changed so I'd need the original to play the replay). 

The thing that bothers me is that I feel I would've noticed my PC deleting 20gb of maps, I was at the pc during the whole thing I explained in my OP. But as far as I can tell the folders are very light, 633gb ish as I said so... ¯\_(ツ)_/¯

But, I'll come back if this happens again in the future ^^ Love your dedication and philosphy towards most things osu! related(especially monetization). Keep being awesome man.

**Reply by peppy** [reply]
Keep in mind the shortcuts don't point to your data folder. The place where you found "backups" should be the correct place though. There should be a "files" folder there which you can check the size of.

**Reply by bearsomehand** [reply]
Yea that one was 800mb something(the whole osu folder that contains the files folder), IIRC. I'm 100% sure it wasn't above 1gb anyway. 

I'm on a fresh install now regardless and playing around with all the new mods ^_^. 
Thanks though.
